netlink-packet-route/src/rtnl/link/nlas/link_infos.rs

```diff
@@ -894,6 +913,62 @@ impl<'a, T: AsRef<[u8]> + ?Sized> Parseable<NlaBuffer<&'a T>> for InfoBridge {
     }
 }
 
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub enum InfoIpoib {
+    Unspec(Vec<u8>),
+    Pkey(u16),
+    Mode(u16),
+    UmCast(u16),
+    Other(DefaultNla),
+}
+
+impl Nla for InfoIpoib {
+    fn value_len(&self) -> usize {
+        use self::InfoIpoib::*;
+        match self {
+            Unspec(bytes) => bytes.len(),
+            Pkey(_) | Mode(_) | UmCast(_) => 2,
+            Other(nla) => nla.value_len(),
+        }
+    }
+
+    fn emit_value(&self, buffer: &mut [u8]) {
+        use self::InfoIpoib::*;
+        match self {
+            Unspec(bytes) => buffer.copy_from_slice(bytes.as_slice()),
+            Pkey(value) => NativeEndian::write_u16(buffer, *value),
+            Mode(value) => NativeEndian::write_u16(buffer, *value),
+            UmCast(value) => NativeEndian::write_u16(buffer, *value),
+            Other(nla) => nla.emit_value(buffer),
+        }
+    }
+
+    fn kind(&self) -> u16 {
+        use self::InfoIpoib::*;
+        match self {
+            Unspec(_) => IFLA_IPOIB_UNSPEC,
+            Pkey(_) => IFLA_IPOIB_PKEY,
+            Mode(_) => IFLA_IPOIB_MODE,
+            UmCast(_) => IFLA_IPOIB_UMCAST,
+            Other(nla) => nla.kind(),
+        }
+    }
+}
+
+impl<'a, T: AsRef<[u8]> + ?Sized> Parseable<NlaBuffer<&'a T>> for InfoIpoib {
+    fn parse(buf: &NlaBuffer<&'a T>) -> Result<Self, DecodeError> {
+        use self::InfoIpoib::*;
+        let payload = buf.value();
+        Ok(match buf.kind() {
+            IFLA_IPOIB_UNSPEC => Unspec(payload.to_vec()),
```

Comment:
This is never used by kernel code. Even when terrible thing happens, we still have `Other(DefaultNla)` to hold it.

Reply:
Well but if the `*_UNSPEC` const exists, shouldn't we have a corresponding enum variant, even if it's not _supposed_ to be created by the kernel? The `Other` variant if more for when there's no known kind. Plus that's how I did it in the rest of the repo so it'd be more consistent with the rest.